mllib/src/main/scala/org/apache/spark/mllib/tree/impurity/Impurity.scala

```diff
@@ -79,7 +79,12 @@ private[spark] abstract class ImpurityAggregator(val statsSize: Int) extends Ser
    * @param allStats  Flat stats array, with stats for this (node, feature, bin) contiguous.
    * @param offset    Start index of stats for this (node, feature, bin).
    */
-  def update(allStats: Array[Double], offset: Int, label: Double, instanceWeight: Double): Unit
+  def update(
+      allStats: Array[Double],
+      offset: Int,
+      label: Double,
+      numSamples: Int,
+      sampleWeight: Double): Unit
```

Comment:
I don't think it's necessary. It's a private class, and the only params currently in the doc are the ambiguous ones. These new ones should be self explanatory.